Add failing test for DDC-3460

flack opened the pull request:
as discussed in #4270 here's my first stab at a failing test.

flack committed:
Add failing testcase for DDC-3460 / gh-4270

Ocramius commented on tests/Doctrine/Tests/ORM/Functional/SchemaTool/DDC3460Test.php:
```diff
@@ -0,0 +1,36 @@
+            $this->markTestSkipped('this test is only relevant for MySQL');
+        }
+
+        $conn->exec('CREATE TABLE `tweet_user` (id INT NOT NULL AUTO_INCREMENT, name VARCHAR(255) NOT NULL, PRIMARY KEY(id)) ENGINE = MyISAM');
```
Add two `DROP TABLE IF EXISTS` before these statements

Ocramius replied:
These models are already in use somewhere else - this makes the test a bit fragile...

flack replied:
Yes, I was lazy :-) I can make new dedicated models for this or I can add a `tearDown` method which drops the tables, if you prefer

Ocramius commented on tests/Doctrine/Tests/ORM/Functional/SchemaTool/DDC3460Test.php:
```diff
@@ -0,0 +1,36 @@
+        $fromSchema = $conn->getSchemaManager()->createSchema();
+
+        $classMetadata = [
+            $this->_em->getClassMetadata('Doctrine\Tests\Models\Tweet\Tweet'),
```
`::class`

flack replied:
I was kind of hoping that this might get backported to 2.5 eventually, so I refrained from using PHP5.5-only stuff. But I can change it ofc

Ocramius replied:
Do make a `master`-compatible one - backporting is up to us :-)

Ocramius commented on tests/Doctrine/Tests/ORM/Functional/SchemaTool/DDC3460Test.php:
```diff
@@ -0,0 +1,36 @@
+
+        $classMetadata = [
+            $this->_em->getClassMetadata('Doctrine\Tests\Models\Tweet\Tweet'),
+            $this->_em->getClassMetadata('Doctrine\Tests\Models\Tweet\User')
```
`::class`

Ocramius commented on tests/Doctrine/Tests/ORM/Functional/SchemaTool/DDC3460Test.php:
```diff
@@ -0,0 +1,36 @@
+
+        $comparator = new \Doctrine\DBAL\Schema\Comparator();
+        $schemaDiff = $comparator->compare($fromSchema, $toSchema);
+        $this->assertEquals([], $schemaDiff->toSql($conn->getDatabasePlatform()));
```
Add a message to the assertion. Also, try checking the existing schema's `engine` attribute on the tables

flack replied:
would this be the correct way to do it? 

```php
$fromSchema->getTable('tweet_user')->getOption('engine')
```

I just tried and it currentyl throws a notice "Undefined index: engine". But that's probably what this test is about :-)

Ocramius replied:
Yes, I think that would be correct.

Ocramius replied:
Well, `self::assertSame('MyISAM', $fromSchema->getTable('tweet_user')->getOption('engine'))`

flack committed:
Address feedback from code review

flack commented:
Puhsed another commit which I think addresses all your points. I've done the `engine` check a bit differently so that it doesn't throw a notice, but still fails. Let me know if I should change something else
